diff -rupN upstream.original/ogr/swq.h upstream.patched/ogr/swq.h
--- upstream.original/ogr/swq.h	2014-09-24 16:47:42.000000000 +0300
+++ upstream.patched/ogr/swq.h	2014-10-02 10:09:32.185143300 +0300
@@ -25,11 +25,13 @@
 #include "cpl_string.h"
 #include "ogr_core.h"
 
+#if !defined(strcasecmp)
 #if defined(_WIN32) && !defined(_WIN32_WCE)
 #  define strcasecmp stricmp
 #elif defined(_WIN32_WCE)
 #  define strcasecmp _stricmp
 #endif
+#endif // !defined(strcasecmp)
 
 typedef enum {
     SWQ_OR,
