diff -rupN upstream.original/frmts/gtiff/gt_citation.cpp upstream.patched/frmts/gtiff/gt_citation.cpp
--- upstream.original/frmts/gtiff/gt_citation.cpp	2014-09-24 17:47:42.000000000 +0400
+++ upstream.patched/frmts/gtiff/gt_citation.cpp	2017-09-26 00:19:19.000000000 +0300
@@ -156,7 +156,7 @@ char* ImagineCitationTranslation(char* p
                 if(p1[0] == '\0' || p1[0] == '\n' || p1[0] == ' ')
                     p1 --;
                 p2 = p1 - 1;
-                while( p2>0 && (p2[0] == ' ' || p2[0] == '\0' || p2[0] == '\n') )
+                while( p2 != 0 && (p2[0] == ' ' || p2[0] == '\0' || p2[0] == '\n') )
                     p2--;
                 if(p2 != p1 - 1)
                     p1 = p2;
@@ -199,7 +199,7 @@ char* ImagineCitationTranslation(char* p
                 if(p1[0] == '\0' || p1[0] == '\n' || p1[0] == ' ')
                     p1 --;
                 char* p2 = p1 - 1;
-                while( p2>0 && (p2[0] == ' ' || p2[0] == '\0' || p2[0] == '\n') )
+                while( p2 != 0 && (p2[0] == ' ' || p2[0] == '\0' || p2[0] == '\n') )
                     p2--;
                 if(p2 != p1 - 1)
                     p1 = p2;
