diff --git a/mkspecs/winrt-arm-msvc2012/qmake.conf b/mkspecs/winrt-arm-msvc2012/qmake.conf
index dfb8674..b7437f0 100644
--- a/mkspecs/winrt-arm-msvc2012/qmake.conf
+++ b/mkspecs/winrt-arm-msvc2012/qmake.conf
@@ -9,6 +9,7 @@ DEFINES                += WINAPI_FAMILY=WINAPI_FAMILY_APP ARM __ARM__ __arm__
 
 QMAKE_LFLAGS           += /MACHINE:ARM
 
+QMAKE_LIBS_CORE         = runtimeobject.lib
 QMAKE_LIBS              = windowscodecs.lib kernel32.lib ole32.lib
 
 VCPROJ_ARCH             = ARM
diff --git a/mkspecs/winrt-x64-msvc2012/qmake.conf b/mkspecs/winrt-x64-msvc2012/qmake.conf
index fb13195..d8b22e1 100644
--- a/mkspecs/winrt-x64-msvc2012/qmake.conf
+++ b/mkspecs/winrt-x64-msvc2012/qmake.conf
@@ -9,6 +9,7 @@ DEFINES                += WINAPI_FAMILY=WINAPI_FAMILY_APP X64 __X64__ __x64__
 
 QMAKE_LFLAGS           += /MACHINE:X64
 
+QMAKE_LIBS_CORE         = runtimeobject.lib
 QMAKE_LIBS              = windowscodecs.lib kernel32.lib ole32.lib
 
 VCPROJ_ARCH             = x64
diff --git a/mkspecs/winrt-x86-msvc2012/qmake.conf b/mkspecs/winrt-x86-msvc2012/qmake.conf
index 288cf02..dcab877 100644
--- a/mkspecs/winrt-x86-msvc2012/qmake.conf
+++ b/mkspecs/winrt-x86-msvc2012/qmake.conf
@@ -9,6 +9,7 @@ DEFINES                += WINAPI_FAMILY=WINAPI_FAMILY_APP X86 __X86__ __x86__
 
 QMAKE_LFLAGS           += /MACHINE:X86
 
+QMAKE_LIBS_CORE         = runtimeobject.lib
 QMAKE_LIBS              = windowscodecs.lib kernel32.lib ole32.lib
 
 VCPROJ_ARCH             = x86
