diff -rupN upstream.patched/source/i18n/rbnf.cpp upstream.original/source/i18n/rbnf.cpp
--- upstream.patched/source/i18n/rbnf.cpp	2014-02-20 22:54:07.062452000 +0200
+++ upstream.original/source/i18n/rbnf.cpp	2013-10-04 23:48:44.000000000 +0300
@@ -31,11 +31,9 @@
 // debugging
 // #define DEBUG
 
-#if 0
 #ifdef DEBUG
 #include "stdio.h"
 #endif
-#endif
 
 #define U_ICUDATA_RBNF U_ICUDATA_NAME U_TREE_SEPARATOR_STRING "rbnf"
 
@@ -326,13 +324,11 @@ private:
     UChar*  nextString(void);
 };
 
-#if 0
 #ifdef DEBUG
 #define ERROR(msg) parseError(msg); return NULL;
 #else
 #define ERROR(msg) parseError(NULL); return NULL;
 #endif
-#endif
         
 
 static const UChar DQUOTE_STOPLIST[] = { 
@@ -555,7 +551,6 @@ LocDataParser::parseError(const char* /*
     pe.postContext[limit-p] = 0;
     pe.offset = (int32_t)(p - data);
     
-#if 0
 #ifdef DEBUG
     fprintf(stderr, "%s at or near character %d: ", str, p-data);
 
@@ -575,7 +570,6 @@ LocDataParser::parseError(const char* /*
     fprintf(stderr, "%s\n", buf);
     fflush(stderr);
 #endif
-#endif
     
     uprv_free(data);
     data = NULL;
