diff -rupN upstream.patched/zconf.h upstream.original/zconf.h
--- upstream.original/zconf.h	2012-12-01 19:46:23.999953500 +0200
+++ upstream.patched/zconf.h	2012-11-22 20:07:46.658881200 +0200
@@ -8,6 +8,10 @@
 #ifndef ZCONF_H
 #define ZCONF_H
 
+//OSMAND{
+#include "../zlibmangle.h"
+//}OSMAND
+
 /*
  * If you *really* need a unique prefix for all types and library functions,
  * compile with -DZ_PREFIX. The "standard" zlib should be compiled without it.
