diff -rupN upstream.original/mkspecs/features/qt_functions.prf upstream.patched/mkspecs/features/qt_functions.prf
--- upstream.original/mkspecs/features/qt_functions.prf	2014-02-06 20:04:09.690123500 +0200
+++ upstream.patched/mkspecs/features/qt_functions.prf	2014-02-06 22:08:52.447836300 +0200
@@ -1,8 +1,5 @@
 defineReplace(qtPlatformTargetSuffix) {
-    ios:CONFIG(iphonesimulator, iphonesimulator|iphoneos): \
-        suffix = _iphonesimulator
-    else: \
-        suffix =
+    suffix =
 
     CONFIG(debug, debug|release) {
         !debug_and_release|build_pass {
diff -rupN upstream.original/mkspecs/macx-ios-clang-device-armv7/qmake.conf upstream.patched/mkspecs/macx-ios-clang-device-armv7/qmake.conf
--- upstream.original/mkspecs/macx-ios-clang-device-armv7/qmake.conf	2014-02-06 20:04:10.746320000 +0200
+++ upstream.patched/mkspecs/macx-ios-clang-device-armv7/qmake.conf	2014-02-06 20:33:25.893092000 +0200
@@ -7,7 +7,7 @@ CONFIG                 += app_bundle red
 QMAKE_INCREMENTAL_STYLE = sublib
 
 QMAKE_MACOSX_DEPLOYMENT_TARGET  =
-QMAKE_IOS_DEPLOYMENT_TARGET     = 5.0
+QMAKE_IOS_DEPLOYMENT_TARGET     = 6.0
 
 INCLUDEPATH            += $$PWD/ios
 DEFINES                += DARWIN_NO_CARBON QT_NO_PRINTER QT_NO_PRINTDIALOG
@@ -16,7 +16,6 @@ DEFINES                += DARWIN_NO_CARB
 QMAKE_IOS_TARGETED_DEVICE_FAMILY = 1,2
 
 QMAKE_IOS_DEVICE_ARCHS = armv7
-QMAKE_IOS_SIMULATOR_ARCHS = i386
 
 include(../common/ios.conf)
 include(../common/gcc-base-mac.conf)
@@ -25,4 +24,9 @@ include(../common/clang-mac.conf)
 include(../common/ios/clang.conf)
 include(../common/ios/qmake.conf)
 
+QMAKE_OBJECTIVE_CFLAGS += -stdlib=libc++
+QMAKE_CFLAGS           += -stdlib=libc++
+QMAKE_CXXFLAGS         += -stdlib=libc++
+QMAKE_LFLAGS           += -stdlib=libc++
+
 load(qt_config)
diff -rupN upstream.original/mkspecs/macx-ios-clang-device-armv7s/qmake.conf upstream.patched/mkspecs/macx-ios-clang-device-armv7s/qmake.conf
--- upstream.original/mkspecs/macx-ios-clang-device-armv7s/qmake.conf	2014-02-06 20:04:10.746320000 +0200
+++ upstream.patched/mkspecs/macx-ios-clang-device-armv7s/qmake.conf	2014-02-06 20:33:34.290169500 +0200
@@ -7,7 +7,7 @@ CONFIG                 += app_bundle red
 QMAKE_INCREMENTAL_STYLE = sublib
 
 QMAKE_MACOSX_DEPLOYMENT_TARGET  =
-QMAKE_IOS_DEPLOYMENT_TARGET     = 5.0
+QMAKE_IOS_DEPLOYMENT_TARGET     = 6.0
 
 INCLUDEPATH            += $$PWD/ios
 DEFINES                += DARWIN_NO_CARBON QT_NO_PRINTER QT_NO_PRINTDIALOG
@@ -15,8 +15,7 @@ DEFINES                += DARWIN_NO_CARB
 # Universal target (iPhone and iPad)
 QMAKE_IOS_TARGETED_DEVICE_FAMILY = 1,2
 
-QMAKE_IOS_DEVICE_ARCHS = armv7
-QMAKE_IOS_SIMULATOR_ARCHS = i386
+QMAKE_IOS_DEVICE_ARCHS = armv7s
 
 include(../common/ios.conf)
 include(../common/gcc-base-mac.conf)
@@ -25,4 +24,9 @@ include(../common/clang-mac.conf)
 include(../common/ios/clang.conf)
 include(../common/ios/qmake.conf)
 
+QMAKE_OBJECTIVE_CFLAGS += -stdlib=libc++
+QMAKE_CFLAGS           += -stdlib=libc++
+QMAKE_CXXFLAGS         += -stdlib=libc++
+QMAKE_LFLAGS           += -stdlib=libc++
+
 load(qt_config)
diff -rupN upstream.original/mkspecs/macx-ios-clang-simulator-i386/qmake.conf upstream.patched/mkspecs/macx-ios-clang-simulator-i386/qmake.conf
--- upstream.original/mkspecs/macx-ios-clang-simulator-i386/qmake.conf	2014-02-06 20:04:10.746320000 +0200
+++ upstream.patched/mkspecs/macx-ios-clang-simulator-i386/qmake.conf	2014-02-06 20:33:47.961369200 +0200
@@ -7,7 +7,7 @@ CONFIG                 += app_bundle red
 QMAKE_INCREMENTAL_STYLE = sublib
 
 QMAKE_MACOSX_DEPLOYMENT_TARGET  =
-QMAKE_IOS_DEPLOYMENT_TARGET     = 5.0
+QMAKE_IOS_DEPLOYMENT_TARGET     = 6.0
 
 INCLUDEPATH            += $$PWD/ios
 DEFINES                += DARWIN_NO_CARBON QT_NO_PRINTER QT_NO_PRINTDIALOG
@@ -15,7 +15,6 @@ DEFINES                += DARWIN_NO_CARB
 # Universal target (iPhone and iPad)
 QMAKE_IOS_TARGETED_DEVICE_FAMILY = 1,2
 
-QMAKE_IOS_DEVICE_ARCHS = armv7
 QMAKE_IOS_SIMULATOR_ARCHS = i386
 
 include(../common/ios.conf)
@@ -25,4 +24,9 @@ include(../common/clang-mac.conf)
 include(../common/ios/clang.conf)
 include(../common/ios/qmake.conf)
 
+QMAKE_OBJECTIVE_CFLAGS += -stdlib=libc++
+QMAKE_CFLAGS           += -stdlib=libc++
+QMAKE_CXXFLAGS         += -stdlib=libc++
+QMAKE_LFLAGS           += -stdlib=libc++
+
 load(qt_config)
