diff --git a/mkspecs/unsupported/macx-ios-clang/qmake.conf b/mkspecs/unsupported/macx-ios-clang/qmake.conf
index 3dc2715..c360b87 100644
--- a/mkspecs/unsupported/macx-ios-clang/qmake.conf
+++ b/mkspecs/unsupported/macx-ios-clang/qmake.conf
@@ -7,7 +7,7 @@ CONFIG                 += app_bundle reduce_exports incremental global_init_link
 QMAKE_INCREMENTAL_STYLE = sublib
 
 QMAKE_MACOSX_DEPLOYMENT_TARGET  =
-QMAKE_IOS_DEPLOYMENT_TARGET     = 4.3
+QMAKE_IOS_DEPLOYMENT_TARGET     = 5.0
 
 INCLUDEPATH            += $$PWD/ios
 DEFINES                += DARWIN_NO_CARBON QT_NO_CORESERVICES QT_NO_PRINTER QT_NO_PRINTDIALOG
@@ -22,4 +22,8 @@ include(../../common/clang-mac.conf)
 include(../../common/ios/clang.conf)
 include(../../common/ios/qmake.conf)
 
+QMAKE_OBJECTIVE_CFLAGS += -stdlib=libc++
+QMAKE_CXXFLAGS         += -stdlib=libc++
+QMAKE_LFLAGS           += -stdlib=libc++
+
 load(qt_config)
