diff -rupN upstream.original/glm/detail/setup.hpp upstream.patched/glm/detail/setup.hpp
--- upstream.original/glm/detail/setup.hpp	2014-04-01 01:01:00.000000000 +0300
+++ upstream.patched/glm/detail/setup.hpp	2014-05-14 20:51:02.403861400 +0300
@@ -543,7 +543,7 @@
 #endif
 
 // Not standard
-#define GLM_HAS_ANONYMOUS_UNION (GLM_LANG & GLM_LANG_CXXMS_FLAG)
+#define GLM_HAS_ANONYMOUS_UNION ((GLM_LANG & GLM_LANG_CXXMS_FLAG) || (GLM_LANG & GLM_LANG_CXX11))
 
 /////////////////
 // Platform 
