diff --git a/contrib/harfbuzz-unicode.h b/contrib/harfbuzz-unicode.h
index f28b3c3..5b290da 100644
--- a/contrib/harfbuzz-unicode.h
+++ b/contrib/harfbuzz-unicode.h
@@ -2,6 +2,13 @@
 #define SCRIPT_IDENTIFY_H_
 
 #include <stdint.h>
+#if defined(_WIN32) && !defined(WINAPI_FAMILY)
+#   ifdef _WIN64
+        typedef __int64    ssize_t;
+#   else
+        typedef _W64 int   ssize_t;
+#   endif
+#endif
 
 #include <harfbuzz-shaper.h>
 
