diff -rupN upstream.original/libs/thread/build/Jamfile.v2 upstream.patched/libs/thread/build/Jamfile.v2
--- upstream.original/libs/thread/build/Jamfile.v2	2018-08-01 23:50:53.000000000 +0300
+++ upstream.patched/libs/thread/build/Jamfile.v2	2018-09-02 09:51:22.000000000 +0300
@@ -147,16 +147,6 @@
 {
     local result = $(name) ;
 
-    if $(type) in STATIC_LIB SHARED_LIB IMPORT_LIB
-    {
-        local api = [ $(property-set).get <threadapi> ] ;
-
-        # non native api gets additional tag
-        if $(api) != [ threadapi-feature.get-default $(property-set) ] {
-            result = $(result)_$(api) ;
-        }
-    }
-
     # forward to the boost tagging rule
     return  [ indirect.call $(BOOST_JAMROOT_MODULE)%$(BOOST_JAMROOT_MODULE).tag
                 $(result) : $(type) : $(property-set) ] ;
