--- upstream.original/mkspecs/macx-ios-clang-device/qmake.conf	2018-09-03 18:22:45.000000000 +0300
+++ upstream.patched/mkspecs/macx-ios-clang-device/qmake.conf	2018-09-03 18:31:47.000000000 +0300
@@ -8,7 +8,6 @@ QMAKE_IOS_DEPLOYMENT_TARGET = 10.0
 QMAKE_APPLE_TARGETED_DEVICE_FAMILY = 1,2
 
 QMAKE_APPLE_DEVICE_ARCHS = arm64
-QMAKE_APPLE_SIMULATOR_ARCHS = x86_64
 
 include(../common/ios.conf)
 include(../common/gcc-base-mac.conf)
