--- upstream.original/include/config/SkUserConfig.h	2015-01-15 17:38:59.280850900 +0200
+++ upstream.patched/include/config/SkUserConfig.h	2015-01-15 17:46:52.425550100 +0200
@@ -178,4 +178,20 @@
 //#define SK_MUTEX_PLATFORM_H "SkMutex_xxx.h"
 //#define SK_BARRIERS_PLATFORM_H "SkBarriers_xxx.h"
 
+/*
+ *  Default to OpenGL/DirectX order
+ *  order (in memory: r,g,b,a)
+ */
+#ifdef SK_CPU_BENDIAN
+    #define SK_R32_SHIFT    24
+    #define SK_G32_SHIFT    16
+    #define SK_B32_SHIFT    8
+    #define SK_A32_SHIFT    0
+#else
+    #define SK_R32_SHIFT    0
+    #define SK_G32_SHIFT    8
+    #define SK_B32_SHIFT    16
+    #define SK_A32_SHIFT    24
+#endif
+
 #endif
