--- upstream.original/src/opts/SkBlitRow_opts_arm_neon.cpp	2014-03-05 18:06:03.150983100 +0200
+++ upstream.patched/src/opts/SkBlitRow_opts_arm_neon.cpp	2014-03-11 23:37:47.850450800 +0200
@@ -129,9 +129,13 @@ void S32A_D565_Opaque_neon(uint16_t* SK_
 
                       "11:                                        \n\t"
                       // unzips achieve the same as a vld4 operation
+#ifndef __clang__
                       "vuzpq.u16  q0, q1                      \n\t"
                       "vuzp.u8    d0, d1                      \n\t"
                       "vuzp.u8    d2, d3                      \n\t"
+#else
+                      "vld4.u8    {d0, d1, d2, d3}, [%[src]]! \n\t"
+#endif
                       // expand 0565 q12 to 8888 {d4-d7}
                       "vmovn.u16  d4, q12                     \n\t"
                       "vshr.u16   q11, q12, #5                \n\t"
