diff -rupN upstream.original/src/ports/SkFontHost_linux.cpp upstream.patched/src/ports/SkFontHost_linux.cpp
--- upstream.original/src/ports/SkFontHost_linux.cpp	2014-08-20 16:57:48.518946300 +0300
+++ upstream.patched/src/ports/SkFontHost_linux.cpp	2014-08-21 09:23:15.675515600 +0300
@@ -268,7 +268,7 @@ class EmptyTypeface : public FamilyTypef
 public:
     EmptyTypeface() : INHERITED(SkTypeface::kNormal, true, NULL, false) {}
 
-    virtual const char* getUniqueString() SK_OVERRIDE const { return NULL; }
+    virtual const char* getUniqueString() const SK_OVERRIDE { return NULL; }
 
 protected:
     virtual SkStream* onOpenStream(int*) const SK_OVERRIDE { return NULL; }
