*** upstream.original/src/core/SkOpts.cpp	2021-11-10 19:03:47.672371100 +0200
--- upstream.patched/src/core/SkOpts.cpp	2021-11-10 19:03:54.978872800 +0200
***************
*** 116,140 ****
  
      static void init() {
      #if defined(SK_CPU_X86)
-         #if SK_CPU_SSE_LEVEL < SK_CPU_SSE_LEVEL_SSSE3
-             if (SkCpu::Supports(SkCpu::SSSE3)) { Init_ssse3(); }
-         #endif
- 
-         #if SK_CPU_SSE_LEVEL < SK_CPU_SSE_LEVEL_SSE42
-             if (SkCpu::Supports(SkCpu::SSE42)) { Init_sse42(); }
-         #endif
- 
-         #if SK_CPU_SSE_LEVEL < SK_CPU_SSE_LEVEL_AVX
-             if (SkCpu::Supports(SkCpu::AVX)) { Init_avx();   }
-             if (SkCpu::Supports(SkCpu::HSW)) { Init_hsw();   }
-         #endif
- 
-         #if SK_CPU_SSE_LEVEL < SK_CPU_SSE_LEVEL_SKX
-             if (SkCpu::Supports(SkCpu::SKX)) { Init_skx(); }
-         #endif
- 
-         if (SkCpu::Supports(SkCpu::ERMS)) { Init_erms(); }
- 
      #elif defined(SK_CPU_ARM64)
          if (SkCpu::Supports(SkCpu::CRC32)) { Init_crc32(); }
  
--- 116,121 ----
