--- upstream.original/src/gpu/gl/win/GrGLCreateNativeInterface_win.cpp	2015-01-15 17:39:01.897001000 +0200
+++ upstream.patched/src/gpu/gl/win/GrGLCreateNativeInterface_win.cpp	2015-01-15 17:46:52.491554000 +0200
@@ -15,7 +15,7 @@
 class AutoLibraryUnload {
 public:
     AutoLibraryUnload(const char* moduleName) {
-        fModule = LoadLibrary(moduleName);
+        fModule = LoadLibraryA(moduleName);
     }
     ~AutoLibraryUnload() {
         if (fModule) {
