--- upstream.original/include/private/SkTHash.h	2023-12-24 15:18:37.908951318 +0500
+++ upstream.patched/include/private/SkTHash.h	2023-12-24 15:27:19.350330038 +0500
@@ -14,6 +14,10 @@
 #include <new>
 #include <utility>
 
+#ifdef Q_FOREACH
+#   undef Q_FOREACH
+#endif
+
 // Before trying to use SkTHashTable, look below to see if SkTHashMap or SkTHashSet works for you.
 // They're easier to use, usually perform the same, and have fewer sharp edges.
 
