diff --git a/src/images/SkImages.cpp b/src/images/SkImages.cpp
index 5b6bf6b..164c8c4 100644
--- a/src/images/SkImages.cpp
+++ b/src/images/SkImages.cpp
@@ -9,13 +9,17 @@
 #include "SkImageRef_GlobalPool.h"
 #include "SkImages.h"
 
+#ifdef OSMAND_ALLOW_ASHMEM
 #ifdef SK_BUILD_FOR_ANDROID
 #include "SkImageRef_ashmem.h"
 #endif
+#endif
 
 void SkImages::InitializeFlattenables() {
     SK_DEFINE_FLATTENABLE_REGISTRAR_ENTRY(SkImageRef_GlobalPool)
+#ifdef OSMAND_ALLOW_ASHMEM
 #ifdef SK_BUILD_FOR_ANDROID
     SK_DEFINE_FLATTENABLE_REGISTRAR_ENTRY(SkImageRef_ashmem)
 #endif
+#endif
 }
