diff --git a/include/config/SkUserConfig.h b/include/config/SkUserConfig.h
index 81033f3..934c2de 100644
--- a/include/config/SkUserConfig.h
+++ b/include/config/SkUserConfig.h
@@ -177,4 +177,20 @@
 //#define SK_ATOMICS_PLATFORM_H "SkAtomics_xxx.h"
 //#define SK_MUTEX_PLATFORM_H "SkMutex_xxx.h"
 
+/*
+ *  Default to OpenGL/DirectX order
+ *  order (in memory: r,g,b,a)
+ */
+#ifdef SK_CPU_BENDIAN
+    #define SK_R32_SHIFT    24
+    #define SK_G32_SHIFT    16
+    #define SK_B32_SHIFT    8
+    #define SK_A32_SHIFT    0
+#else
+    #define SK_R32_SHIFT    0
+    #define SK_G32_SHIFT    8
+    #define SK_B32_SHIFT    16
+    #define SK_A32_SHIFT    24
+#endif
+
 #endif
