diff --git a/src/ports/SkFontHost_android.cpp b/src/ports/SkFontHost_android.cpp
index 12357f1..e50a7f3 100644
--- a/src/ports/SkFontHost_android.cpp
+++ b/src/ports/SkFontHost_android.cpp
@@ -883,7 +883,8 @@ size_t SkFontHost::GetFileName(SkFontID fontID, char path[], size_t length,
 }

 SkFontID SkFontHost::NextLogicalFont(SkFontID currFontID, SkFontID origFontID) {
-#ifdef SK_BUILD_FOR_ANDROID_NDK
+#undef OSMAND_NO_FALLBACK_FONTS
+#ifdef OSMAND_NO_FALLBACK_FONTS
     // Skia does not support font fallback for ndk applications in order to
     // enable clients such as WebKit to customize their font selection.
     // Clients can use GetFallbackFamilyNameForChar() to get the fallback
