diff --git a/src/gpu/gl/win/GrGLCreateNativeInterface_win.cpp b/src/gpu/gl/win/GrGLCreateNativeInterface_win.cpp
index 6adaf19..5e5b0a3 100644
--- a/src/gpu/gl/win/GrGLCreateNativeInterface_win.cpp
+++ b/src/gpu/gl/win/GrGLCreateNativeInterface_win.cpp
@@ -14,7 +14,7 @@
 class AutoLibraryUnload {
 public:
     AutoLibraryUnload(const char* moduleName) {
-        fModule = LoadLibrary(moduleName);
+        fModule = LoadLibraryA(moduleName);
     }
     ~AutoLibraryUnload() {
         if (NULL != fModule) {
