--- upstream.original/src/utils/win/SkWGL_win.cpp	2015-01-15 17:39:02.708044300 +0200
+++ upstream.patched/src/utils/win/SkWGL_win.cpp	2015-01-15 17:46:52.457552400 +0200
@@ -224,7 +224,7 @@ void destroy_dummy_window(HWND dummy) {
 }
 
 #define GET_PROC(NAME, SUFFIX) f##NAME = \
-                     (##NAME##Proc) wglGetProcAddress("wgl" #NAME #SUFFIX)
+                     ( NAME##Proc ) wglGetProcAddress("wgl" #NAME #SUFFIX)
 
 SkWGLExtensions::SkWGLExtensions()
     : fGetExtensionsString(NULL)
