diff -rupN upstream.original/src/opts/SkUtils_opts_arm.cpp upstream.patched/src/opts/SkUtils_opts_arm.cpp
--- upstream.original/src/opts/SkUtils_opts_arm.cpp	2015-01-15 17:39:02.120010800 +0200
+++ upstream.patched/src/opts/SkUtils_opts_arm.cpp	2015-01-15 18:16:28.403156100 +0200
@@ -19,8 +19,7 @@ extern "C" void arm_memset32(uint32_t* d
 #endif
 
 SkMemset16Proc SkMemset16GetPlatformProc() {
-    // FIXME: memset.arm.S is using syntax incompatible with XCode
-#if !defined(SK_CPU_LENDIAN) || defined(SK_BUILD_FOR_IOS)
+#if !defined(SK_CPU_LENDIAN)
     return NULL;
 #elif SK_ARM_NEON_IS_DYNAMIC
     if (sk_cpu_arm_has_neon()) {
@@ -36,8 +35,7 @@ SkMemset16Proc SkMemset16GetPlatformProc
 }
 
 SkMemset32Proc SkMemset32GetPlatformProc() {
-    // FIXME: memset.arm.S is using syntax incompatible with XCode
-#if !defined(SK_CPU_LENDIAN) || defined(SK_BUILD_FOR_IOS)
+#if !defined(SK_CPU_LENDIAN)
     return NULL;
 #elif SK_ARM_NEON_IS_DYNAMIC
     if (sk_cpu_arm_has_neon()) {
diff -rupN upstream.original/src/opts/arm-assember.h upstream.patched/src/opts/arm-assember.h
--- upstream.original/src/opts/arm-assember.h	1970-01-01 02:00:00.000000000 +0200
+++ upstream.patched/src/opts/arm-assember.h	2015-01-15 18:24:48.663914600 +0200
@@ -0,0 +1,19 @@
+#ifdef __APPLE__
+#   define ENTRY(x) _##x
+#   define GLOBAL(x) .globl _##x
+#   define UNWIND_START
+#   define UNWIND_END
+#   define FUNC_START
+#   define FUNC_END
+#   define THE_END
+#   define USE_FPU(x)
+#else
+#   define ENTRY(x) x
+#   define GLOBAL(x) .global x; .type x, %function
+#   define UNWIND_START .fnstart
+#   define UNWIND_END .fnend
+#   define FUNC_START .func
+#   define FUNC_END .endfunc
+#   define THE_END .end
+#   define USE_FPU(x) .fpu x
+#endif
diff -rupN upstream.original/src/opts/memset.arm.S upstream.patched/src/opts/memset.arm.S
--- upstream.original/src/opts/memset.arm.S	2015-01-15 17:39:02.138012100 +0200
+++ upstream.patched/src/opts/memset.arm.S	2015-01-15 18:19:33.275569400 +0200
@@ -5,6 +5,8 @@
  * found in the LICENSE file.
  */
 
+#include "arm-assember.h"
+
 /* Changes:
  * 2010-08-11 Steve McIntyre <steve.mcintyre@arm.com>
  *    Added small changes to the two functions to make them work on the
@@ -17,10 +19,8 @@
     .align 4
     .syntax unified
 
-    .global arm_memset32
-    .type   arm_memset32, %function
-    .global arm_memset16
-    .type   arm_memset16, %function
+    GLOBAL(arm_memset32)
+    GLOBAL(arm_memset16)
 
 /*
  * Optimized memset functions for ARM.
@@ -29,8 +29,8 @@
  * void arm_memset32(uint32_t* dst, uint32_t value, int count);
  *
  */
-arm_memset16:
-        .fnstart
+ENTRY(arm_memset16):
+        UNWIND_START
         push        {lr}
 
         /* if count is equal to zero then abort */
@@ -51,10 +51,10 @@ arm_memset16:
 
         /* Now jump into the main loop below. */
         b           .Lwork_32
-        .fnend
+        UNWIND_END
 
-arm_memset32:
-        .fnstart
+ENTRY(arm_memset32):
+        UNWIND_START
         push        {lr}
 
         /* if count is equal to zero then abort */
@@ -108,4 +108,4 @@ arm_memset32:
 
 .Lfinish:
         pop         {pc}
-        .fnend
+        UNWIND_END
--- upstream.original/src/opts/memset16_neon.S	2015-01-16 11:26:22.502494700 +0200
+++ upstream.patched/src/opts/memset16_neon.S	2015-01-17 23:12:29.504343100 +0200
@@ -5,6 +5,8 @@
  * found in the LICENSE file.
  ***************************************************************************/
 
+#include "arm-assember.h"
+
 /***************************************************************************
   Neon memset: Attempts to do a memset with Neon registers if possible,
      Inputs:
@@ -18,11 +20,11 @@
         .syntax unified
 
         .code 32
-        .fpu neon
+        USE_FPU(neon)
         .align 4
-        .globl memset16_neon
+        GLOBAL(memset16_neon)
 
-memset16_neon:
+ENTRY(memset16_neon):
         cmp             r2, #0
         bxeq            lr
 
@@ -140,4 +142,4 @@ memset_end:
         pop             {r0}
         bx              lr
 
-        .end
+        THE_END
diff -rupN upstream.original/src/opts/memset32_neon.S upstream.patched/src/opts/memset32_neon.S
--- upstream.original/src/opts/memset32_neon.S	2015-01-15 17:39:02.141012200 +0200
+++ upstream.patched/src/opts/memset32_neon.S	2015-01-15 18:23:53.983011700 +0200
@@ -5,13 +5,15 @@
  * found in the LICENSE file.
  ***************************************************************************/
 
+#include "arm-assember.h"
+
 	.code 32
-	.fpu neon
+	USE_FPU(neon)
 	.align 4
-	.globl	memset32_neon
+	GLOBAL(memset32_neon)
 
 	/* r0 = buffer, r1 = value, r2 = times to write */
-memset32_neon:
+ENTRY(memset32_neon):
 	cmp		r2, #1
 	streq		r1, [r0], #4
 	bxeq		lr
@@ -108,4 +110,4 @@ memset32_dropthru:
 	str		r1, [r0, #0]
 	bx		lr
 
-	.end
+	THE_END
