--- upstream.original/include/core/SkPreConfig.h	2015-01-15 17:38:59.399858900 +0200
+++ upstream.patched/include/core/SkPreConfig.h	2015-01-15 18:31:07.972068700 +0200
@@ -197,6 +198,10 @@
     #define SK_CPU_ARM64
 #endif
 
+#if !defined(SK_ARM_HAS_NEON) && defined(__ARM_NEON__)
+    #define SK_ARM_HAS_NEON
+#endif
+
 //////////////////////////////////////////////////////////////////////
 
 #if !defined(SKIA_IMPLEMENTATION)
