diff -rupN upstream.original/src/gpu/gl/android/GrGLCreateNativeInterface_android.cpp upstream.patched/src/gpu/gl/android/GrGLCreateNativeInterface_android.cpp
--- upstream.original/src/gpu/gl/android/GrGLCreateNativeInterface_android.cpp	2013-09-27 17:22:47.392743700 +0300
+++ upstream.patched/src/gpu/gl/android/GrGLCreateNativeInterface_android.cpp	2013-09-27 23:23:00.950665100 +0300
@@ -31,7 +31,12 @@ static const GrGLInterface* create_es_in
     interface->fBindAttribLocation = glBindAttribLocation;
     interface->fBindBuffer = glBindBuffer;
     interface->fBindTexture = glBindTexture;
+#if defined(__ANDROID_API__) && __ANDROID_API__ >= 9
     interface->fBindVertexArray = glBindVertexArrayOES;
+#else
+    interface->fBindVertexArray =
+        (GrGLBindVertexArrayProc) eglGetProcAddress("glBindVertexArrayOES");
+#endif
     interface->fBlendColor = glBlendColor;
     interface->fBlendFunc = glBlendFunc;
     interface->fBufferData = glBufferData;
@@ -50,7 +55,12 @@ static const GrGLInterface* create_es_in
     interface->fDeleteProgram = glDeleteProgram;
     interface->fDeleteShader = glDeleteShader;
     interface->fDeleteTextures = glDeleteTextures;
+#if defined(__ANDROID_API__) && __ANDROID_API__ >= 9
     interface->fDeleteVertexArrays = glDeleteVertexArraysOES;
+#else
+    interface->fDeleteVertexArrays =
+        (GrGLDeleteVertexArraysProc) eglGetProcAddress("glDeleteVertexArraysOES");
+#endif
     interface->fDepthMask = glDepthMask;
     interface->fDisable = glDisable;
     interface->fDisableVertexAttribArray = glDisableVertexAttribArray;
@@ -64,7 +74,12 @@ static const GrGLInterface* create_es_in
     interface->fGenBuffers = glGenBuffers;
     interface->fGenerateMipmap = glGenerateMipmap;
     interface->fGenTextures = glGenTextures;
+#if defined(__ANDROID_API__) && __ANDROID_API__ >= 9
     interface->fGenVertexArrays = glGenVertexArraysOES;
+#else
+    interface->fGenVertexArrays =
+        (GrGLGenVertexArraysProc) eglGetProcAddress("glGenVertexArraysOES");
+#endif
     interface->fGetBufferParameteriv = glGetBufferParameteriv;
     interface->fGetError = glGetError;
     interface->fGetIntegerv = glGetIntegerv;
@@ -114,8 +129,11 @@ static const GrGLInterface* create_es_in
 #endif
     }
 
-#if GL_EXT_discard_framebuffer
+#if GL_EXT_discard_framebuffer && (defined(__ANDROID_API__) && __ANDROID_API__ >= 9)
     interface->fDiscardFramebuffer = glDiscardFramebufferEXT;
+#else
+    interface->fDiscardFramebuffer =
+        (GrGLDiscardFramebufferProc) eglGetProcAddress("glDiscardFramebufferEXT");
 #endif
     interface->fUniform1f = glUniform1f;
     interface->fUniform1i = glUniform1i;
@@ -150,7 +168,7 @@ static const GrGLInterface* create_es_in
 #if GR_GL_IGNORE_ES3_MSAA
 
     if (extensions.has("GL_EXT_multisampled_render_to_texture")) {
-#if GL_EXT_multisampled_render_to_texture
+#if GL_EXT_multisampled_render_to_texture && (defined(__ANDROID_API__) && __ANDROID_API__ >= 9)
         interface->fFramebufferTexture2DMultisample = glFramebufferTexture2DMultisampleEXT;
         interface->fRenderbufferStorageMultisample = glRenderbufferStorageMultisampleEXT;
 #else
@@ -158,7 +176,7 @@ static const GrGLInterface* create_es_in
         interface->fRenderbufferStorageMultisample = (GrGLRenderbufferStorageMultisampleProc) eglGetProcAddress("glRenderbufferStorageMultisampleEXT");
 #endif
     } else if (extensions.has("GL_IMG_multisampled_render_to_texture")) {
-#if GL_IMG_multisampled_render_to_texture
+#if GL_IMG_multisampled_render_to_texture && (defined(__ANDROID_API__) && __ANDROID_API__ >= 9)
         interface->fFramebufferTexture2DMultisample = glFramebufferTexture2DMultisampleIMG;
         interface->fRenderbufferStorageMultisample = glRenderbufferStorageMultisampleIMG;
 #else
@@ -179,7 +197,7 @@ static const GrGLInterface* create_es_in
 #endif
         }
         if (extensions.has("GL_EXT_multisampled_render_to_texture")) {
-#if GL_EXT_multisampled_render_to_texture
+#if GL_EXT_multisampled_render_to_texture && (defined(__ANDROID_API__) && __ANDROID_API__ >= 9)
             interface->fFramebufferTexture2DMultisample = glFramebufferTexture2DMultisampleEXT;
             interface->fRenderbufferStorageMultisampleES2EXT = glRenderbufferStorageMultisampleEXT;
 #else
@@ -187,7 +205,7 @@ static const GrGLInterface* create_es_in
             interface->fRenderbufferStorageMultisampleES2EXT = (GrGLRenderbufferStorageMultisampleProc) eglGetProcAddress("glRenderbufferStorageMultisampleEXT");
 #endif
         } else if (extensions.has("GL_IMG_multisampled_render_to_texture")) {
-#if GL_IMG_multisampled_render_to_texture
+#if GL_IMG_multisampled_render_to_texture && (defined(__ANDROID_API__) && __ANDROID_API__ >= 9)
             interface->fFramebufferTexture2DMultisample = glFramebufferTexture2DMultisampleIMG;
             interface->fRenderbufferStorageMultisampleES2EXT = glRenderbufferStorageMultisampleIMG;
 #else
@@ -202,7 +220,7 @@ static const GrGLInterface* create_es_in
     interface->fGetFramebufferAttachmentParameteriv = glGetFramebufferAttachmentParameteriv;
     interface->fGetRenderbufferParameteriv = glGetRenderbufferParameteriv;
     interface->fRenderbufferStorage = glRenderbufferStorage;
-#if GL_OES_mapbuffer
+#if GL_OES_mapbuffer && (defined(__ANDROID_API__) && __ANDROID_API__ >= 9)
     interface->fMapBuffer = glMapBufferOES;
     interface->fUnmapBuffer = glUnmapBufferOES;
 #else
