diff -rupN upstream.original/libarchive/archive_entry.h upstream.patched/libarchive/archive_entry.h
--- upstream.original/libarchive/archive_entry.h	2022-01-22 10:56:13.831237254 +0100
+++ upstream.patched/libarchive/archive_entry.h	2022-01-23 09:36:24.884197067 +0100
@@ -99,7 +99,7 @@ typedef ssize_t la_ssize_t;
 #endif
 
 /* Large file support for Android */
-#ifdef __ANDROID__
+#if defined(__LIBARCHIVE_BUILD) && defined(__ANDROID__)
 #include "android_lf.h"
 #endif
 
diff -rupN upstream.original/libarchive/archive.h upstream.patched/libarchive/archive.h
--- upstream.original/libarchive/archive.h	2022-01-22 10:56:13.827237176 +0100
+++ upstream.patched/libarchive/archive.h	2022-01-23 09:35:57.567668270 +0100
@@ -97,7 +97,7 @@ typedef ssize_t la_ssize_t;
 #endif
 
 /* Large file support for Android */
-#ifdef __ANDROID__
+#if defined(__LIBARCHIVE_BUILD) && defined(__ANDROID__)
 #include "android_lf.h"
 #endif
 
