diff -rupN upstream.original/libarchive/archive_match.c upstream.patched/libarchive/archive_match.c
--- upstream.original/libarchive/archive_match.c	2013-02-08 02:52:07.000000000 +0200
+++ upstream.patched/libarchive/archive_match.c	2014-04-19 20:50:00.020276000 +0300
@@ -1152,7 +1152,7 @@ set_timefilter_pathname_mbs(struct archi
 {
 	/* NOTE: stat() on Windows cannot handle nano seconds. */
 	HANDLE h;
-	WIN32_FIND_DATA d;
+	WIN32_FIND_DATAA d;
 
 	if (path == NULL || *path == '\0') {
 		archive_set_error(&(a->archive), EINVAL, "pathname is empty");
diff -rupN upstream.original/libarchive/archive_write_disk_windows.c upstream.patched/libarchive/archive_write_disk_windows.c
--- upstream.original/libarchive/archive_write_disk_windows.c	2013-01-14 03:43:45.000000000 +0200
+++ upstream.patched/libarchive/archive_write_disk_windows.c	2014-04-19 20:50:37.579038700 +0300
@@ -525,7 +525,7 @@ la_GetFunctionKernel32(const char *name)
 	static int set;
 	if (!set) {
 		set = 1;
-		lib = LoadLibrary("kernel32.dll");
+		lib = LoadLibraryA("kernel32.dll");
 	}
 	if (lib == NULL) {
 		fprintf(stderr, "Can't load kernel32.dll?!\n");
diff -rupN upstream.original/libarchive/filter_fork_windows.c upstream.patched/libarchive/filter_fork_windows.c
--- upstream.original/libarchive/filter_fork_windows.c	2013-01-14 03:43:45.000000000 +0200
+++ upstream.patched/libarchive/filter_fork_windows.c	2014-04-19 20:50:47.571183600 +0300
@@ -36,7 +36,7 @@ __archive_create_child(const char *cmd,
 {
 	HANDLE childStdout[2], childStdin[2],childStderr;
 	SECURITY_ATTRIBUTES secAtts;
-	STARTUPINFO staInfo;
+	STARTUPINFOA staInfo;
 	PROCESS_INFORMATION childInfo;
 	struct archive_string cmdline;
 	struct archive_string fullpath;
