diff --git a/mkspecs/macx-clang-libc++/qmake.conf b/mkspecs/macx-clang-libc++/qmake.conf
index 94cc003..c72434d 100644
--- a/mkspecs/macx-clang-libc++/qmake.conf
+++ b/mkspecs/macx-clang-libc++/qmake.conf
@@ -17,4 +17,9 @@ QMAKE_OBJECTIVE_CFLAGS += -stdlib=libc++
 QMAKE_CXXFLAGS         += -stdlib=libc++
 QMAKE_LFLAGS           += -stdlib=libc++
 
+QMAKE_CFLAGS           += -arch i386 -arch x86_64
+QMAKE_OBJECTIVE_CFLAGS += -arch i386 -arch x86_64
+QMAKE_CXXFLAGS         += -arch i386 -arch x86_64
+QMAKE_LFLAGS           += -arch i386 -arch x86_64
+
 load(qt_config)
