diff --git a/mkspecs/macx-clang-libc++-64/qmake.conf b/mkspecs/macx-clang-libc++-64/qmake.conf
new file mode 100644
index 0000000..ce37cbb
--- /dev/null
+++ b/mkspecs/macx-clang-libc++-64/qmake.conf
@@ -0,0 +1,21 @@
+#
+# qmake configuration for 64-bit Clang on OS X
+#
+
+MAKEFILE_GENERATOR      = UNIX
+CONFIG                 += app_bundle incremental global_init_link_order lib_version_first plugin_no_soname
+QMAKE_INCREMENTAL_STYLE = sublib
+
+include(../common/macx.conf)
+include(../common/gcc-base-mac.conf)
+include(../common/clang.conf)
+include(../common/clang-mac.conf)
+
+QMAKE_MACOSX_DEPLOYMENT_TARGET = 10.7
+
+QMAKE_CFLAGS           += -arch x86_64
+QMAKE_OBJECTIVE_CFLAGS += -arch x86_64 -stdlib=libc++
+QMAKE_CXXFLAGS         += -arch x86_64 -stdlib=libc++
+QMAKE_LFLAGS           += -arch x86_64 -stdlib=libc++
+
+load(qt_config)
diff --git a/mkspecs/macx-clang-libc++-64/qplatformdefs.h b/mkspecs/macx-clang-libc++-64/qplatformdefs.h
new file mode 100644
index 0000000..2c3a8b1
--- /dev/null
+++ b/mkspecs/macx-clang-libc++-64/qplatformdefs.h
@@ -0,0 +1,43 @@
+/****************************************************************************
+**
+** Copyright (C) 2013 Digia Plc and/or its subsidiary(-ies).
+** Contact: http://www.qt-project.org/legal
+**
+** This file is part of the qmake spec of the Qt Toolkit.
+**
+** $QT_BEGIN_LICENSE:LGPL$
+** Commercial License Usage
+** Licensees holding valid commercial Qt licenses may use this file in
+** accordance with the commercial license agreement provided with the
+** Software or, alternatively, in accordance with the terms contained in
+** a written agreement between you and Digia.  For licensing terms and
+** conditions see http://qt.digia.com/licensing.  For further information
+** use the contact form at http://qt.digia.com/contact-us.
+**
+** GNU Lesser General Public License Usage
+** Alternatively, this file may be used under the terms of the GNU Lesser
+** General Public License version 2.1 as published by the Free Software
+** Foundation and appearing in the file LICENSE.LGPL included in the
+** packaging of this file.  Please review the following information to
+** ensure the GNU Lesser General Public License version 2.1 requirements
+** will be met: http://www.gnu.org/licenses/old-licenses/lgpl-2.1.html.
+**
+** In addition, as a special exception, Digia gives you certain additional
+** rights.  These rights are described in the Digia Qt LGPL Exception
+** version 1.1, included in the file LGPL_EXCEPTION.txt in this package.
+**
+** GNU General Public License Usage
+** Alternatively, this file may be used under the terms of the GNU
+** General Public License version 3.0 as published by the Free Software
+** Foundation and appearing in the file LICENSE.GPL included in the
+** packaging of this file.  Please review the following information to
+** ensure the GNU General Public License version 3.0 requirements will be
+** met: http://www.gnu.org/copyleft/gpl.html.
+**
+**
+** $QT_END_LICENSE$
+**
+****************************************************************************/
+
+#include "../common/mac/qplatformdefs.h"
+
