diff -rupN upstream.original/mkspecs/linux-clang-32/qmake.conf upstream.patched/mkspecs/linux-clang-32/qmake.conf
--- upstream.original/mkspecs/linux-clang-32/qmake.conf	2015-02-28 15:07:03.481509000 +0200
+++ upstream.patched/mkspecs/linux-clang-32/qmake.conf	2015-02-28 15:23:50.165061700 +0200
@@ -7,6 +7,9 @@ CONFIG += incremental
 
 QMAKE_INCREMENTAL_STYLE = sublib
 
+QMAKE_CFLAGS = -m32
+QMAKE_LFLAGS = -m32
+
 include(../common/linux.conf)
 include(../common/gcc-base-unix.conf)
 include(../common/clang.conf)
diff -rupN upstream.original/mkspecs/linux-clang-64/qmake.conf upstream.patched/mkspecs/linux-clang-64/qmake.conf
--- upstream.original/mkspecs/linux-clang-64/qmake.conf	2015-02-28 15:07:03.481509000 +0200
+++ upstream.patched/mkspecs/linux-clang-64/qmake.conf	2015-02-28 15:23:50.170065500 +0200
@@ -7,6 +7,9 @@ CONFIG += incremental
 
 QMAKE_INCREMENTAL_STYLE = sublib
 
+QMAKE_CFLAGS = -m64
+QMAKE_LFLAGS = -m64
+
 include(../common/linux.conf)
 include(../common/gcc-base-unix.conf)
 include(../common/clang.conf)
