--- upstream.original/mkspecs/macx-clang-libc++-32/qmake.conf	2016-07-21 20:17:26.000000000 +0300
+++ upstream.patched/mkspecs/macx-clang-libc++-32/qmake.conf	2016-07-21 20:20:38.000000000 +0300
@@ -13,4 +13,9 @@ include(../common/clang-mac.conf)
 
 QMAKE_MACOSX_DEPLOYMENT_TARGET = 10.8
 
+QMAKE_CFLAGS           += -arch i386
+QMAKE_OBJECTIVE_CFLAGS += -arch i386 -stdlib=libc++
+QMAKE_CXXFLAGS         += -arch i386 -stdlib=libc++
+QMAKE_LFLAGS           += -arch i386 -stdlib=libc++
+
 load(qt_config)
--- upstream.original/mkspecs/macx-clang-libc++-64/qmake.conf	2016-07-21 20:17:26.000000000 +0300
+++ upstream.patched/mkspecs/macx-clang-libc++-64/qmake.conf	2016-07-21 20:20:38.000000000 +0300
@@ -13,4 +13,9 @@ include(../common/clang-mac.conf)
 
 QMAKE_MACOSX_DEPLOYMENT_TARGET = 10.8
 
+QMAKE_CFLAGS           += -arch x86_64
+QMAKE_OBJECTIVE_CFLAGS += -arch x86_64 -stdlib=libc++
+QMAKE_CXXFLAGS         += -arch x86_64 -stdlib=libc++
+QMAKE_LFLAGS           += -arch x86_64 -stdlib=libc++
+
 load(qt_config)
