diff -rupN upstream.original/src/corelib/arch/qatomic_armv5.h upstream.patched/src/corelib/arch/qatomic_armv5.h
--- upstream.original/src/corelib/arch/qatomic_armv5.h	2015-03-01 18:02:31.050284400 +0200
+++ upstream.patched/src/corelib/arch/qatomic_armv5.h	2015-03-01 18:06:02.171905200 +0200
@@ -61,9 +61,6 @@ QT_END_NAMESPACE
 #ifdef QT_NO_ARM_EABI
 # error "Sorry, ARM without EABI is no longer supported"
 #endif
-#ifndef Q_OS_LINUX
-# error "Qt is misconfigured: this ARMv5 implementation is only possible on Linux"
-#endif
 
 template <int size> struct QBasicAtomicOps: QGenericAtomicOps<QBasicAtomicOps<size> >
 {
