diff --git a/src/corelib/io/qfsfileengine_unix.cpp b/src/corelib/io/qfsfileengine_unix.cpp
index a5f077b..d0286ec 100644
--- a/src/corelib/io/qfsfileengine_unix.cpp
+++ b/src/corelib/io/qfsfileengine_unix.cpp
@@ -263,6 +263,9 @@ bool QFSFileEnginePrivate::nativeFlush()
 bool QFSFileEnginePrivate::nativeSyncToDisk()
 {
     Q_Q(QFSFileEngine);
+#if defined(Q_OS_ANDROID) && defined(__ANDROID_API__) && __ANDROID_API__ < 9
+# undef _POSIX_SYNCHRONIZED_IO
+#endif
 #if defined(_POSIX_SYNCHRONIZED_IO) && _POSIX_SYNCHRONIZED_IO > 0
     const int ret = fdatasync(nativeHandle());
 #else
