diff --git a/src/corelib/arch/qatomic_armv5.h b/src/corelib/arch/qatomic_armv5.h
index e0a50b3..2307a9f 100644
--- a/src/corelib/arch/qatomic_armv5.h
+++ b/src/corelib/arch/qatomic_armv5.h
@@ -68,9 +68,6 @@ QT_END_NAMESPACE
 #ifdef QT_NO_ARM_EABI
 # error "Sorry, ARM without EABI is no longer supported"
 #endif
-#ifndef Q_OS_LINUX
-# error "Qt is misconfigured: this ARMv5 implementation is only possible on Linux"
-#endif
 
 template<> struct QAtomicIntegerTraits<int> { enum { IsInteger = 1 }; };
 template<> struct QAtomicIntegerTraits<unsigned int> { enum { IsInteger = 1 }; };
