diff -rupN upstream.original/src/corelib/kernel/qjni.cpp upstream.patched/src/corelib/kernel/qjni.cpp
--- upstream.original/src/corelib/kernel/qjni.cpp	2014-08-26 10:14:30.635564300 +0300
+++ upstream.patched/src/corelib/kernel/qjni.cpp	2014-08-26 14:10:17.316619400 +0300
@@ -1264,7 +1264,7 @@ jdouble QJNIObjectPrivate::callStaticMet
     return callStaticMethod<jdouble>(clazz, methodName, "()D");
 }
 
-QJNIObjectPrivate QJNIObjectPrivate::callObjectMethod(const char *methodName,
+QJNIObjectPrivate QJNIObjectPrivate::callObjectMethod_VA(const char *methodName,
                                                       const char *sig,
                                                       va_list args) const
 {
@@ -1288,7 +1288,7 @@ QJNIObjectPrivate QJNIObjectPrivate::cal
 {
     va_list args;
     va_start(args, sig);
-    QJNIObjectPrivate res = callObjectMethod(methodName, sig, args);
+    QJNIObjectPrivate res = callObjectMethod_VA(methodName, sig, args);
     va_end(args);
     return res;
 }
@@ -1341,7 +1341,7 @@ QJNIObjectPrivate QJNIObjectPrivate::cal
     return callObjectMethod(methodName, "()[D");
 }
 
-QJNIObjectPrivate QJNIObjectPrivate::callStaticObjectMethod(const char *className,
+QJNIObjectPrivate QJNIObjectPrivate::callStaticObjectMethod_VA(const char *className,
                                                             const char *methodName,
                                                             const char *sig,
                                                             va_list args)
@@ -1370,12 +1370,12 @@ QJNIObjectPrivate QJNIObjectPrivate::cal
 {
     va_list args;
     va_start(args, sig);
-    QJNIObjectPrivate res = callStaticObjectMethod(className, methodName, sig, args);
+    QJNIObjectPrivate res = callStaticObjectMethod_VA(className, methodName, sig, args);
     va_end(args);
     return res;
 }
 
-QJNIObjectPrivate QJNIObjectPrivate::callStaticObjectMethod(jclass clazz,
+QJNIObjectPrivate QJNIObjectPrivate::callStaticObjectMethod_VA(jclass clazz,
                                                             const char *methodName,
                                                             const char *sig,
                                                             va_list args)
@@ -1401,7 +1401,7 @@ QJNIObjectPrivate QJNIObjectPrivate::cal
 {
     va_list args;
     va_start(args, sig);
-    QJNIObjectPrivate res = callStaticObjectMethod(clazz, methodName, sig, args);
+    QJNIObjectPrivate res = callStaticObjectMethod_VA(clazz, methodName, sig, args);
     va_end(args);
     return res;
 }
diff -rupN upstream.original/src/corelib/kernel/qjni_p.h upstream.patched/src/corelib/kernel/qjni_p.h
--- upstream.original/src/corelib/kernel/qjni_p.h	2014-08-26 10:14:30.636540300 +0300
+++ upstream.patched/src/corelib/kernel/qjni_p.h	2014-08-26 14:10:20.229000100 +0300
@@ -200,7 +200,7 @@ private:
     T callMethod(const char *methodName,
                  const char *sig,
                  va_list args) const;
-    QJNIObjectPrivate callObjectMethod(const char *methodName,
+    QJNIObjectPrivate callObjectMethod_VA(const char *methodName,
                                        const char *sig,
                                        va_list args) const;
     template <typename T>
@@ -211,11 +211,11 @@ private:
     static T callStaticMethod(jclass clazz,
                               const char *methodName,
                               const char *sig, va_list args);
-    static QJNIObjectPrivate callStaticObjectMethod(const char *className,
+    static QJNIObjectPrivate callStaticObjectMethod_VA(const char *className,
                                                     const char *methodName,
                                                     const char *sig, va_list args);
 
-    static QJNIObjectPrivate callStaticObjectMethod(jclass clazz,
+    static QJNIObjectPrivate callStaticObjectMethod_VA(jclass clazz,
                                                     const char *methodName,
                                                     const char *sig, va_list args);
 
