diff -rupN upstream.original/mkspecs/win32-g++/qmake.conf upstream.patched/mkspecs/win32-g++/qmake.conf
--- upstream.original/mkspecs/win32-g++/qmake.conf	2015-03-01 18:02:27.642005300 +0200
+++ upstream.patched/mkspecs/win32-g++/qmake.conf	2015-03-01 18:10:00.243665600 +0200
@@ -67,7 +67,7 @@ QMAKE_RUN_CXX_IMP       = $(CXX) -c $(CX
 
 QMAKE_LINK              = $${CROSS_COMPILE}g++
 QMAKE_LINK_C            = $${CROSS_COMPILE}gcc
-QMAKE_LFLAGS            =
+QMAKE_LFLAGS            = -static-libgcc -static-libstdc++
 QMAKE_LFLAGS_EXCEPTIONS_ON = -mthreads
 QMAKE_LFLAGS_EXCEPTIONS_OFF =
 QMAKE_LFLAGS_RELEASE    = -Wl,-s
