--- upstream.original/XSB/emu/system_xsb.c	2018-02-09 19:11:30.000000000 +0300
+++ upstream.patched/XSB/emu/system_xsb.c	2018-02-09 18:18:53.000000000 +0300
@@ -275,7 +275,7 @@
   switch (callno) {
   case PLAIN_SYSTEM_CALL: /* dumb system call: no communication with XSB */
     /* this call is superseded by shell and isn't used */
-    ctop_int(CTXTc 3, system(ptoc_string(CTXTc 2)));
+    //ctop_int(CTXTc 3, system(ptoc_string(CTXTc 2)));
     return TRUE;
   case SLEEP_FOR_SECS:
 #ifdef WIN_NT
@@ -810,7 +810,7 @@
   } else { /* SHELL command */
     /* no separator */
     concat_array(CTXTc argv, "", shell_command, MAX_CMD_LEN);
-    pid = system(shell_command);
+    //pid = system(shell_command);
   }
 
   /* main process continues */
