diff -rupN upstream.original/png.h upstream.patched/png.h
--- upstream.original/png.h	2013-07-18 03:02:59.000000000 +0300
+++ upstream.patched/png.h	2013-09-09 08:38:41.919951800 +0300
@@ -443,6 +443,10 @@
 #ifndef PNG_VERSION_INFO_ONLY
    /* Machine specific configuration. */
 #  include "pngconf.h"
+
+//OSMAND{
+#include "../libpngmangle.h"
+//}OSMAND
 #endif
 
 /*
diff -rupN upstream.original/pnglibconf.h upstream.patched/pnglibconf.h
--- upstream.original/pnglibconf.h	1970-01-01 02:00:00.000000000 +0200
+++ upstream.patched/pnglibconf.h	2013-09-09 08:38:41.935965800 +0300
@@ -0,0 +1 @@
+#include "scripts/pnglibconf.h.prebuilt"
